Scripts/DowloadPortalItemData.swift

```diff
@@ -125,61 +126,50 @@ func uncompressArchive(at sourceURL: URL, to destinationURL: URL) throws {
-///   - completion: A closure to handle the results.
-func downloadFile(at sourceURL: URL, to downloadDirectory: URL, completion: @escaping (Result<URL, Error>) -> Void) {
-    let downloadTaskCompleted = { (temporaryURL: URL?, response: URLResponse?, error: Error?) in
-        if let temporaryURL = temporaryURL,
-           let response = response,
-           let suggestedFilename = response.suggestedFilename {
-            do {
-                let downloadName: String
-                let isArchive = (suggestedFilename as NSString).pathExtension == "zip"
-                // If the downloaded file is an archive and contains
-                //   - 1 file, use the name of that file.
-                //   - multiple files, use the suggested filename (*.zip).
-                // If it is not an archive, use the server suggested filename.
-                if isArchive {
-                    let count = try count(ofFilesInArchiveAt: temporaryURL)
-                    if count == 1 {
-                        downloadName = try name(ofFileInArchiveAt: temporaryURL)
-                    } else {
-                        downloadName = suggestedFilename
-                    }
-                } else {
-                    downloadName = suggestedFilename
-                }
-                
-                let downloadURL = downloadDirectory.appendingPathComponent(downloadName, isDirectory: false)
-                
-                if FileManager.default.fileExists(atPath: downloadURL.path) {
-                    try FileManager.default.removeItem(at: downloadURL)
-                }
-                
-                if isArchive {
-                    let extractURL = downloadURL.pathExtension == "zip"
-                        // Uncompresses to directory named after archive.
-                        ? downloadURL.deletingPathExtension()
-                        // Uncompresses to appropriate subdirectory.
-                        : downloadURL.deletingLastPathComponent()
-                    try uncompressArchive(at: temporaryURL, to: extractURL)
-                } else {
-                    try FileManager.default.moveItem(at: temporaryURL, to: downloadURL)
-                }
-                
-                completion(.success(downloadURL))
-            } catch {
-                completion(.failure(error))
-            }
-        } else if let error = error {
-            completion(.failure(error))
+///   - downloadDirectory: The directory to store the downloaded data in.
+/// - Throws: Exceptions when downloading and moving the file.
+/// - Returns: The name of the downloaded file.
+func downloadFile(from sourceURL: URL, to downloadDirectory: URL) async throws -> String {
+    let portalURLRequest = URLRequest(url: sourceURL)
+    let (temporaryURL, response) = try await URLSession.shared.download(for: portalURLRequest)
+    
+    guard let suggestedFilename = response.suggestedFilename else { return "" }
```

Comment:
This seems to be a behavioral change. The old script omits the case of no `suggestedFilename` and no error. Here it returns an empty string.